myceliald/src/http_client.rs

```diff
@@ -35,43 +36,79 @@ fn is_for_client(config: &Config, name: &str) -> bool {
-        let client_token = config.server.token.clone();
-
+    fn new(
+        config: ClientConfig,
+        scheduler_handle: SchedulerHandle,
+        storage_handle: SqliteStorageHandle,
+    ) -> Self {
         Self {
             config,
-            client_token,
             scheduler_handle,
+            storage_handle,
         }
     }
 
+    // Client should register only once. Subsequently, it should use the client_id and client_secret it gets back from the registration to authenticate itself.
+    async fn register_if_not_registered(&mut self) -> Result<(), SectionError> {
+        let state = self
+            .storage_handle
+            .retrieve_state(std::u64::MAX)
+            .await
+            .unwrap();
+        if state.is_none() {
+            return self.register().await;
+        }
+        let state = state.unwrap();
+        let client_id: Option<String> = state.get("auth0_client_id").unwrap();
+        let client_secret: Option<String> = state.get("auth0_client_secret").unwrap();
+        if client_id.is_none() || client_secret.is_none() {
+            return self.register().await;
+        }
+        // also store these values so they can be injected into the mycelial network sections
+        if client_id.is_some() && client_secret.is_some() {
+            self.scheduler_handle
+                .set_client_id_and_secret(client_id.unwrap(), client_secret.unwrap())
+                .await?;
+        }
+        Ok(())
+    }
+
+    // Client should register only once. Subsequently, it should use the client_id and client_secret it gets back from the registration to authenticate itself.
     async fn register(&mut self) -> Result<(), SectionError> {
         let client = reqwest::Client::new();
         let url = format!("{}/api/client", self.config.server.endpoint.as_str());
-        let _x: ProvisionClientResponse = client
+        let resp = client
             .post(url)
             .header("Authorization", self.basic_auth())
             .json(&ProvisionClientRequest {
                 client_config: self.config.clone(),
             })
             .send()
-            .await?
-            .json()
             .await?;
-
-        let url = format!("{}/api/tokens", self.config.server.endpoint.as_str());
-        let token: IssueTokenResponse = client
-            .post(url)
-            .header("Authorization", self.basic_auth())
-            .json(&IssueTokenRequest {
-                client_id: self.config.node.unique_id.clone(),
-            })
-            .send()
-            .await?
-            .json()
+        if resp.status() != 200 {
+            return Err(format!(
+                "failed to register client - status code: {:?}",
+                resp.status()
+            )
+            .into());
+        }
+        let provision_client_resp: ProvisionClientResponse = resp.json().await?;
+
+        let mut state = SqliteState::new();
```

Comment:
http client should have own state management, mixing state of two different things is not a good approach here.